.slashrc

```diff
@@ -208,6 +208,13 @@ class MediaPlugin(slash.plugins.PluginInterface):
     from lib.platform import info
     return info()["gpu"]["gen"]
 
+  def _get_expected_mfx_runtime(self, dispatcher):
+    if "msdk" == dispatcher:
+      return "msdk" if self._get_gpu_gen() < 12.2 else "vpl"
+    if "vpl" == dispatcher:
+      return "msdk" if self._get_gpu_gen() < 12 else "vpl"
```

Comment:
Only one dispatcher can be linked to the middleware at compile time (msdk=libmfx.so, vpl=libvpl.so).  If we run ffmpeg-qsv or gst-msdk plugins, then they will load the "linked" dispatcher.  This is used to determine which runtime should be loaded by the dispatcher.

Reply:
In CI, we use "ffmpeg-libvpl" to indicate ffmpeg-qsv linked against libvpl (vpl dispatcher).  And "ffmpeg-libmfx" to indicate ffmpeg-qsv linked against libmfx (msdk dispatcher).

Reply:
In other words, all devices will support both msdk (legacy) and vpl dispatchers.  But only one dispatcher can be linked to the application/middleware at a time.  And it is the dispatchers job to determine which runtime to load on that device.  And the expected runtimes depending on dispatcher linked is a fixed POR.

Reply:
Please read the commit message for https://github.com/intel/vaapi-fits/pull/306/commits/e3ee57d80f902715d5667407aa5894f38b483e09  Maybe it is more clear?

Reply:
> I know it is the 'dispatchers job', but here our return is our expect value of 'run time'
> So it looks like we expect when gen >= 12.2 we expect it is "vpl", this is not the problem
> but when gen = 12 we couldn't expect which run time be chosen by dispatcher, because both support
> from the CI, it also been install together,
> maybe it cause by dispatch detect run time initialize success
> it also means for the gen = 12 we will miss 'vpl dispatcher' + 'msdk run time' on the test scope

The vpl dispatcher will never load msdk runtime on gen 12 (+).  This is the POR, otherwise it is a test "failure".

Reply:
Please refer to the PXT tables that I forwarded to you.  Those are POR.

Reply:
@xhaihao @FocusLuo comment please?  Does it make sense?

Reply:
> like this example: TGL support libvpl and both run time, so here will get 'vpl' ?

Sorry, I missed this comment.  Yes, TGL (and other gen 12.0) support both runtimes.  But msdk and vpl dispatcher always choose the different runtime.  This is the only overlap... to ease customer transition time.

> OK, I'm understand for this line, it will fix in feature, the vpl dispatcher on gen 12(+) will just load vpl runtime

Yep, the loaded runtime is fixed in dispatcher.  Hope we're all clear now. :smiley: